src/_pytest/_code/code.py

```diff
@@ -633,13 +633,13 @@ def getrepr(
         )
         return fmt.repr_excinfo(self)
 
-    def match(self, regexp: "Union[str, Pattern]") -> bool:
+    def match(self, regexp: "Union[str, Pattern]") -> "Literal[True]":
         """
-        Check whether the regular expression 'regexp' is found in the string
-        representation of the exception using ``re.search``. If it matches
-        then True is returned (so that it is possible to write
-        ``assert excinfo.match()``). If it doesn't match an AssertionError is
-        raised.
+        Check whether the regular expression `regexp` matches the string
+        representation of the exception using :func:`python:re.search`.
+        If it matches `True` is returned (so that it is possible to write
+        :code:`assert excinfo.match()`).
+        If it doesn't match an ``AssertionError`` is raised.
         """
         __tracebackhide__ = True
         if not re.search(regexp, str(self.value)):
```

Comment:
Will the assertion message improve if the condition is moved to the assert? I mean

```py
assert re.search(regexp, str(self.value)), "Pattern {!r} not found in {!r}".format(regexp, str(self.value))
```

Reply:
Shouldn't make a difference.
What I've meant in the comment above is that "pattern does not match" reads better than "pattern not found in" (which sounds like a "needle in stack" check).

Reply:
If it doesn't make a difference I would change it anyway, seems to me `assert condition` is preferable to `if not condition: assert 0`.